### PR TITLE
Allow badger v1 + v2 and logloadingvaluemode

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: badgerV2/badger.go [hunk 1]
```diff
@@ -0,0 +1,404 @@
+package badger
```
Comment (maraino) [inline badgerV2/badger.go:1]: This file should be `badger/v2/badger.go`

File: nosql.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package nosql
 import (
 	"github.com/pkg/errors"
 	"github.com/smallstep/nosql/badger"
+	badgerV2 "github.com/smallstep/nosql/badgerV2"
```
Comment (maraino) [inline nosql.go:6]: Do those two import compile in all supported architecture? If they do not, we need to split this file by architecture, see https://golang.org/pkg/go/build/#hdr-Build_Constraints

File: database/database.go [hunk 3]
```diff
@@ -53,8 +53,9 @@ func cause(err error) error {
 
 // Options are configuration options for the database.
 type Options struct {
-	Database string
-	ValueDir string
+	Database                  string
+	ValueDir                  string
+	BadgerValueLogLoadingMode string
```
Comment (maraino) [inline database/database.go:58]: In badger this is called `FileLoadingMode`, it should probably be `BadgerFileLoadingMode` or just `FileLoadingMode`, this option will be ignored by the rest of the implementations.

Comment (dopey) [reply database/database.go:58]: In Badger the type is called `FileLoadingMode`, but the attribute is called `ValueLogLoadingMode` in the Badger Options. There are multiple attributes that have the type `FileLoadingMode`. 

```
type Options struct {
	Dir      string
	ValueDir string

	SyncWrites          bool
	TableLoadingMode    options.FileLoadingMode
	ValueLogLoadingMode options.FileLoadingMode
	NumVersionsToKeep   int
	ReadOnly            bool
	Truncate            bool
...
```

File: badger/badger.go [hunk 4]
```diff
@@ -24,6 +26,18 @@ func (db *DB) Open(dir string, opt ...database.Option) (err error) {
 	}
 
 	bo := badger.DefaultOptions
+
+	// Set the ValueLogLoadingMode - default is MemoryMap. Low memory/RAM
+	// systems may want to use FileIO.
+	switch strings.ToLower(opts.BadgerValueLogLoadingMode) {
+	case "", "memorymap":
```
[the remark this answers is not in the document]
Comment (dopey) [reply badger/badger.go:33]:
```
// ErrInvalidLoadingMode is returned when opt.ValueLogLoadingMode option is not
	// within the valid range
	ErrInvalidLoadingMode = errors.New("Invalid ValueLogLoadingMode, must be FileIO or MemoryMap")
```

According to the documentation at pkg.go.dev `ValueLogLoadingMode does not accept the `LoadToRAM` value.

File: database/database.go (hunk 3 above)
Comment (maraino) [reply database/database.go:58]: We should use the same mode for both. In badger Table and Value mean something, but in nosql they don't really if we don't take into account that `ValueDir` option that was added back in the day. Unless we want to define independent values for Table and Value we should just call it FileLoadingMode.

Comment (dopey) [reply database/database.go:58]: I changed it to `FileLoadingMode`. But if we need to go back later and define different values for Table and ValueLog I'm gonna blame you.

File: nosql.go (hunk 2 above)
Comment (dopey) [reply nosql.go:6]: I made darwin, linux, arm64, and armv7 binaries of `certificates` using this code. Looks like all our supported architectures are ok. Anything else I should try?

File: badger/v1/badger.go [hunk 5]
```diff
@@ -1,10 +1,12 @@
-package badger
+package v1
```
Comment (maraino) [inline badger/v1/badger.go:1]: I believe you can call this badger.

File: badger/v2/badger.go [hunk 6]
```diff
@@ -0,0 +1,404 @@
+package v2
```
Comment (maraino) [inline badger/v2/badger.go:1]: This can be badger too

File: nosql.go (hunk 2 above)
Comment (maraino) [reply nosql.go]: No it should be ok.

**Pull request merged**